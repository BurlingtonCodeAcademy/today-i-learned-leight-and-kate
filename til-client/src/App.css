body {
  margin: 0;
  padding: 0;
  background-color: palevioletred;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 2s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  color: #fff;
  height: auto;
  padding: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

p {
  color: #000000;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.Entry-box {
  background-color: white;
  color: #000;
  font-size: 26px;
  font-weight: bold;
  font-family: "Times New Roman", Times, serif;
  margin: 30px 25%;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: white solid 2px;
  border-radius: 5px;
}

.Entry-time {
  color: gray;
  font-size: 16px;
  font-weight: normal;
  font-style: italic;
}
.Entry-author {
  color: gray;
  font-size: 20px;
  font-weight: normal;
  font-style: italic;
}
.Entry-title {
  align-self: flex-start;
  font-size: 28px;
}
.Entry-body {
  color: palevioletred;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 24px;
  font-weight: normal;
  font-style: normal;
}

form {
  width: 100%;
  padding: 0px;
  font-size: 18px;
}
form > input,
textarea {
  font-size: 18px;
}
input {
  width: 150px;
  padding: 5px;
  margin: 5px;
}
#grid-container {
  display: grid;
  justify-items: center;
  grid-template-areas:
    "title body"
    "author body"
    "button button";
}
#author {
  grid-area: author;
  justify-self: end;
  border-radius: 5px;
  border: 0px;
}
#title {
  grid-area: title;
  justify-self: end;
  border-radius: 5px;
  border: 0px;
}
textarea {
  grid-area: body;
  justify-self: start;
  height: 70px;
  width: 150px;
  padding: 5px;
  margin: 5px;
  border-radius: 5px;
  border: 0px;
}

.button {
  grid-area: button;
  width: 10%;
  padding: 8px;
  margin: 10px;
  background-color: palevioletred;
  border: 0px;
  border-radius: 5px;
  color: white;
  font-size: 15px;
  font-weight: bold;
}
.button, .small-button:hover {
  background-color: white;
  color: palevioletred;
}
.small-button {
  grid-area: button;
  width: 50px;
  padding: 8px;
  margin: 10px;
  background-color: palevioletred;
  border: 0px;
  border-radius: 5px;
  color: white;
  font-size: 12px;
}

.buttonArea {
  width: 100%;
}

.status {
  color: white;
  margin: 10px;
}
