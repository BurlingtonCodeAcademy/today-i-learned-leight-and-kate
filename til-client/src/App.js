import React, { Component } from "react";
import Entry from "./Entry.js";
import "./App.css";

class App extends Component {
  constructor() {
    super();
    this.state = { entries: [], author: "", title: "", body: "" };
    const BCA = "http://10.1.10.58";
    // const LOCAL = "http://localhost";
    this.PATH = BCA;
  }

  componentDidMount() {
    fetch(`${this.PATH}:5000/facts`)
      .then(response => response.json())
      .then(data => this.setState({ entries: data }))
      .catch(() => this.setState({ status: "Failed to fetch content" }));
  }

  componentDidUpdate() {
    setTimeout(() => this.setState({ status: "" }), 5000);
  }

  handleChange = event =>
    this.setState({ [event.target.name]: event.target.value });

  handleSubmit = event => {
    event.preventDefault();
    const { author, title, body, entries } = this.state;
    if (!title.trim())
      return this.setState({ status: "TIL must have a title" });
    fetch(`${this.PATH}:5000/facts`, {
      method: "POST",
      mode: "cors",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify({ author: author, title: title, body: body })
    })
      .then(() => {
        entries.unshift({ author: author, title, body });
        this.setState({
          author: "",
          title: "",
          body: "",
          entries,
          status: "Successfully Posted!"
        });
      })
      .catch(() => this.setState({ status: "Failed to post content" }));
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Today I Learned</h1>
          <form id="grid-container" onSubmit={this.handleSubmit}>
            <input
              type="text"
              name="title"
              autoComplete="off"
              placeholder="title"
              id="title"
              value={this.state.title}
              onChange={this.handleChange}
            />
            <input
              type="text"
              name="author"
              placeholder="author"
              id="author"
              value={this.state.author}
              onChange={this.handleChange}
            />
            <textarea
              name="body"
              autoComplete="off"
              placeholder="body"
              value={this.state.body}
              onChange={this.handleChange}
            />
            <input type="submit" value="Post" className="button" />
          </form>
        </header>
        <div>{this.state.status}</div>
        {this.state.entries.map(entry => (
          <Entry key={entry._id} {...entry} />
        ))}
      </div>
    );
  }
}

export default App;
