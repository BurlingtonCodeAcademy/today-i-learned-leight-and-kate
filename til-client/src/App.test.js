import React from "react";
import { mount, shallow } from "enzyme";
import App from "./App";

describe("Using the App", () => {
  it("Renders without crashing", () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });
  const app = mount(<App />);
  it("Renders Form Component", () => {
    expect(app.find("Form").exists()).toBeTruthy();
  });
  it("Form has title, author, body inputs", () => {
    const form = app.find("Form");
    expect(form.find("input[id='title']").exists()).toBeTruthy();
    expect(form.find("input[id='author']").exists()).toBeTruthy();
    expect(form.find("textarea[id='body']").exists()).toBeTruthy();
  });
  afterAll(() => {
    app.unmount();
  });
});

// expect(app.find("Entry").exists()).toBeTruthy();
