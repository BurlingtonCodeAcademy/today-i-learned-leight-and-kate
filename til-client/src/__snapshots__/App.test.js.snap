// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using the App Renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <h1>
            Today I Learned
          </h1>
          <Form
            author=""
            body=""
            buttonText="Post"
            entries={Array []}
            handleChange={[Function]}
            handleSubmit={[Function]}
            title=""
          />
        </header>,
        <div
          className="status"
        />,
        Array [],
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Today I Learned
            </h1>,
            <Form
              author=""
              body=""
              buttonText="Post"
              entries={Array []}
              handleChange={[Function]}
              handleSubmit={[Function]}
              title=""
            />,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Today I Learned",
            },
            "ref": null,
            "rendered": "Today I Learned",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "author": "",
              "body": "",
              "buttonText": "Post",
              "entries": Array [],
              "handleChange": [Function],
              "handleSubmit": [Function],
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "status",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <h1>
              Today I Learned
            </h1>
            <Form
              author=""
              body=""
              buttonText="Post"
              entries={Array []}
              handleChange={[Function]}
              handleSubmit={[Function]}
              title=""
            />
          </header>,
          <div
            className="status"
          />,
          Array [],
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Today I Learned
              </h1>,
              <Form
                author=""
                body=""
                buttonText="Post"
                entries={Array []}
                handleChange={[Function]}
                handleSubmit={[Function]}
                title=""
              />,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Today I Learned",
              },
              "ref": null,
              "rendered": "Today I Learned",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "author": "",
                "body": "",
                "buttonText": "Post",
                "entries": Array [],
                "handleChange": [Function],
                "handleSubmit": [Function],
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "status",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
